common/src/main/java/com/proximyst/ban/service/MessageService.java

```diff
@@ -0,0 +1,230 @@
+//
+// ban - A punishment suite for Velocity.
+// Copyright (C) 2020 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package com.proximyst.ban.service;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.proximyst.ban.config.MessageKey;
+import com.proximyst.ban.config.MessagesConfig;
+import com.proximyst.ban.factory.IBanExceptionalFutureLoggerFactory;
+import com.proximyst.ban.factory.IMessageFactory;
+import com.proximyst.ban.message.IMessage;
+import com.proximyst.ban.message.IMessageComponent;
+import com.proximyst.ban.model.BanUser;
+import com.proximyst.ban.model.Punishment;
+import com.proximyst.ban.platform.IBanAudience;
+import com.proximyst.ban.platform.IBanServer;
+import com.proximyst.ban.utils.BanExceptionalFutureLogger;
+import java.text.SimpleDateFormat;
+import java.util.concurrent.CompletableFuture;
+import net.kyori.adventure.identity.Identity;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+// The existence of this class is because of this issue:
+// <https://github.com/google/guice/issues/1345>.
+@Singleton
+public final class MessageService {
+  private static final @NonNull String KEY_TARGET_NAME = "targetName";
+  private static final @NonNull String KEY_TARGET_UUID = "targetUuid";
+  private static final @NonNull String KEY_PUNISHER_NAME = "punisherName";
+  private static final @NonNull String KEY_PUNISHER_UUID = "punisherUuid";
+  private static final @NonNull String KEY_PUNISHMENT_ID = "punishmentId";
+  private static final @NonNull String KEY_PUNISHMENT_DATE = "punishmentDate";
+  private static final @NonNull String KEY_PUNISHMENT_REASON = "reason";
+  private static final @NonNull String KEY_PUNISHMENT_EXPIRY = "expiry";
+  private static final @NonNull String KEY_PUNISHMENT_DURATION = "duration";
+  private static final @NonNull String KEY_PUNISHMENT_TYPE = "punishmentType";
+  private static final @NonNull String KEY_PUNISHMENT_VERB = "punishmentVerb";
+  private static final @NonNull String KEY_QUANTITY = "amount";
```

Comment:
`@NonNull`  should be redundant for constants, unless checkerframework is being annoying.

Reply:
I plan on using checkerframework's checkers when they support Java 15, in which case they will be annoying.